### PR TITLE
Fix tokenlist gen

### PR DESCRIPTION
**Pull request by fredlacs**

### CODE REVIEW

File: src/lib/token_list_gen.ts [hunk 1]
```diff
@@ -101,9 +111,14 @@ export const generateTokenList = async (
       .map((token, i: number) => {
     const l2GatewayAddress = token.token.joinTableEntry[0].gateway.gatewayAddr;
     let { name:_name, decimals, symbol:_symbol } = token.tokenDatum;
-    if(!_name) throw new Error(`Unexpected undefined token name: ${JSON.stringify(token)}`);
-    if(!decimals) throw new Error(`Unexpected undefined token decimals: ${JSON.stringify(token)}`);
-    if(!_symbol) throw new Error(`Unexpected undefined token symbol: ${JSON.stringify(token)}`);
+    
+    if(_name === undefined || _name === null) throw new Error(`Unexpected undefined token name: ${JSON.stringify(token)}`);
```
Comment (yahgwai) [inline src/lib/token_list_gen.ts:115]: If you want to, this:
```
_name === undefined || _name === null
```
is the same as:
```
_name == undefined
```

But it's fine as it is as well

File: src/lib/token_list_gen.ts [hunk 2]
```diff
@@ -26,21 +26,31 @@ export interface ArbificationOptions {
 export interface L2ToL1GatewayAddresses {
   [contractAddress: string]: string;
 }
+
+// TODO: read these values from the gateway or a subgraph
```
Comment (yahgwai) [inline src/lib/token_list_gen.ts:30]: Maybe delete this and create a ticket in notion instead for better visibility.

Comment (fredlacs) [reply src/lib/token_list_gen.ts:30]: I made the ticket before the TODO tag ;)

File: src/lib/token_list_gen.ts [hunk 3]
```diff
@@ -101,9 +111,28 @@ export const generateTokenList = async (
       .map((token, i: number) => {
     const l2GatewayAddress = token.token.joinTableEntry[0].gateway.gatewayAddr;
     let { name:_name, decimals, symbol:_symbol } = token.tokenDatum;
-    if(!_name) throw new Error(`Unexpected undefined token name: ${JSON.stringify(token)}`);
-    if(!decimals) throw new Error(`Unexpected undefined token decimals: ${JSON.stringify(token)}`);
-    if(!_symbol) throw new Error(`Unexpected undefined token symbol: ${JSON.stringify(token)}`);
+    
+    if(decimals === undefined) throw new Error(`Unexpected undefined token decimals: ${JSON.stringify(token)}`);
```
Comment (yahgwai) [inline src/lib/token_list_gen.ts:115]: Could _name, _decimals or _symbol be null? I dont know if they can, just pointing out that before we would throw an error for that whereas now we wouldnt

Comment (fredlacs) [reply src/lib/token_list_gen.ts:115]: it can't be null according to typescript inference

Comment (fredlacs) [reply src/lib/token_list_gen.ts:115]: the previous issue is that !decimals was handling the value zero as falsey in a token

**Review by yahgwai — APPROVED** [review]

**Pull request merged**